sdk/tests/js/tests/texture-corner-case-videos.js

```diff
@@ -187,6 +187,20 @@ function generateTest(desc,
             debug("Checking upper left corner");
             wtu.checkCanvasRect(gl, 4, gl.canvas.height - 8, 2, 2, topColor,
                                 "shouldBe " + topColor, tolerance);
+
+            // Expose bug http://crbug.com/733172.
+            if (sourceSubRectangle) {
+                // Skip sub-rectangle tests for cube map textures for the moment.
+                if (bindingTarget == gl.TEXTURE_CUBE_MAP) {
+                    continue;
+                }
+                gl.texSubImage2D(targets[tt], 0, 0, 0,
+                                 sourceSubRectangle[2], sourceSubRectangle[3],
+                                 gl[pixelFormat], gl[pixelType], videoElement);
+            } else {
+                gl.texSubImage2D(targets[tt], 0, 0, 0, gl[pixelFormat], gl[pixelType], videoElement);
+            }
+            wtu.glErrorShouldBe(gl, gl.NO_ERROR, "should be no errors");
```

Comment:
No, it won't catch this bug. The error is because we use line 152  'gl.texImage2D(targets[tt], 0, gl[internalFormat], gl[pixelFormat], gl[pixelType], videoElement)' to initialize the texture. In chromium, that API will go gpu-gpu path which will upload the video visible size data. But TexSubImage will call GLContext()->TexSubImage which use video nature size.   But this video has natural size > visible size. So when call texture->ValidForTexture in TextureManager::ValidateTexSubImage, it will not meet the condition  max_x <= info.width and print bad dimentions.

line 128 TexImage2D and TexSubImage2D both use nature size. So it can't catch this error.